scripts/lib/sign-macos.ts

```diff
@@ -249,15 +249,27 @@ async function *findFilesToSign(dir: string): AsyncIterable<string> {
 /**
  * Detect if the path of a plain file indicates that it's the bundle executable
  */
-function isBundleExecutable(fullPath: string): boolean {
+async function isBundleExecutable(fullPath: string): Promise<boolean> {
   const parts = fullPath.split(path.sep).reverse();
 
   if (parts.length >= 4) {
-    // Foo.app/Contents/MacOS/Foo - the check style here avoids spell checker.
-    if (fullPath.endsWith(`${ parts[0] }.app/Contents/MacOS/${ parts[0] }`)) {
-      return true;
+    // Anything.app/Contents/MacOS/executable - the check style here avoids spell checker.
+    if (fullPath.endsWith(`.app/Contents/MacOS/${ parts[0] }`)) {
+      // Check Anything.app/Contents/Info.plist for CFBundleExecutable
+      const infoPlist = path.sep + path.join(...parts.slice(2).reverse(), 'Info.plist');
+
+      try {
+        const { stdout } = await spawnFile('/usr/bin/defaults', ['read', infoPlist, 'CFBundleExecutable'], { stdio: 'pipe' });
```

Comment:
Using `/usr/bin/defaults` to read plist files is deprecated; per its man page:
> WARNING: The defaults command will be changed in an upcoming major release to only operate on preferences domains. General plist manipulation utilities will be folded into a different command-line program.

Please use `plutil -extract CFBundleExecutable raw -expect string foo.plist` instead.

Reply:
Sure